### ISSUE TITLE
Protege unable to re-open a Turtle file

### ISSUE THREAD

**Issue by csnyulas** [protegeproject/protege#595]
**Reported from mailing list:**
Apparently Protégé is unable to load a certain ontology previously created with Protégé and saved in Turtle format. The Turtle parser gives an error message that does not make much sense (to me):
```
org.openrdf.rio.RDFParseException: Expected '.', found ''' [line 59]
```

**Original message:**
_Hi,
I was wondering if you could help me to figure out how to reopen an owl file that I created in Protégé and then saved using the Turtle format. When I try to reopen the file Protégé reports a number of errors and the file is not opened._

_Having checked the content of the owl file it seems to be fine, at least according to my understanding of Turtle, but I could very well be mistaken._

_Can you advise me whether there is there any way that I can reopen the file and recover what I was working on?_

_I've attached the file to this email for your reference._

[PlaysLinesPlaces.owl.zip](https://github.com/protegeproject/protege/files/647460/PlaysLinesPlaces.owl.zip)

This may be an [OWL API](https://github.com/owlcs/owlapi) issue, so I copy @ignazio1977.

**Comment by ignazio1977** [comment]
These two lines appear to show an IRI with a quote character in it:
`
`:BF_I.i.10 rdf:type owl:NamedIndividual , doco:Line ;`
`           <http://schema.org/contentLocation> :St_Paul's_Cathedral`

Looks like bad rendering of an unusual IRI. As a workaround, avoid quotes while we determine where the bug is.

**Comment by csnyulas** [comment]
Good catch!!!
I removed the all 5 `'` characters in the document, and Protégé could open it without problem. The funny thing is that the characters were nowhere near line 59, so the exception message was not very helpful.

I will keep this issue open for now, until we decide if we will do something (and if yes, what) about avoiding these problematic serializations in the future.

**Comment by csnyulas** [comment]
Ooohh, I see now what have I overlooked earlier: I saw only now that in the error message there were 3 `'`s one after another. I saw only two when I first looked at the problem, and the "empty" string did not make much sense.

**Comment by ignazio1977** [comment]
Does not happen in version 4.5.5

**Comment by matthewhorridge** [comment]
@csnyulas please can you check this to see if the problem is fixed in 5.5.0-beta-8?

**Comment by csnyulas** [comment]
This can be considered fixed in latest 5.5.0 beta.

Although the original test file still can't be opened, because it was not properly serialized in previous version of Protege, if we would make the same ontology with the latest version of Protege, and would use single quotes in IRIs, the new Turtle serialization would write it out so that it can be re-opened later.

For the record, here is the output of a diff between how the new turtle file, created with the latest version of Protege, differs form the original file that was problematic: 
```
83c83
<            <http://schema.org/contentLocation> :St_Paul's_Cathedral .
---
>            <http://schema.org/contentLocation> <http://www.semanticweb.org/andy/ontologies/2016/11/untitled-ontology-21#St_Paul's_Cathedral> .
175,177c175,177
< :St_Paul's_Cathedral rdf:type owl:NamedIndividual ,
<                               <http://dbpedia.org/ontology/Place> ;
<                      <http://xmlns.com/foaf/0.1/name> "St Paul's Cathedral"^^xsd:string .
---
> <http://www.semanticweb.org/andy/ontologies/2016/11/untitled-ontology-21#St_Paul's_Cathedral> rdf:type owl:NamedIndividual ,
>                                                                                                        <http://dbpedia.org/ontology/Place> ;
>                                                                                               <http://xmlns.com/foaf/0.1/name> "St Paul's Cathedral"^^xsd:string .
216c216
<                         :St_Paul's_Cathedral
---
>                         <http://www.semanticweb.org/andy/ontologies/2016/11/untitled-ontology-21#St_Paul's_Cathedral>
230c230
< ###  Generated by the OWL API (version 4.2.6.20160910-2108) https://github.com/owlcs/owlapi
---
> ###  Generated by the OWL API (version 4.5.7.2018-12-02T02:23:35Z) https://github.com/owlcs/owlapi
``` 

Note: If someone has a file that was serialized incorrectly with previous versions of Protege, they will need to fix it manually (e.g. in a text editor) in order to be able to work with it in Protege.
